TutorLizard.BusinessLogic/Services/TutorService.cs

```diff
@@ -5,28 +5,73 @@
 using TutorLizard.BusinessLogic.Models.DTOs;
 using TutorLizard.BusinessLogic.Models.DTOs.Requests;
 using TutorLizard.BusinessLogic.Models.DTOs.Responses;
+using TutorLizard.BusinessLogic.Interfaces.Repositories;
+using TutorLizard.BusinessLogic.Models;
+using TutorLizard.BusinessLogic.Interfaces.Data.Repositories;
+using Microsoft.EntityFrameworkCore;
+using System.Net.NetworkInformation;
+using System;
 
 namespace TutorLizard.BusinessLogic.Services;
 public class TutorService : ITutorService
 {
-    private readonly IDbRepository<ScheduleItem> _scheduleItemRepository;
     private readonly IDbRepository<ScheduleItemRequest> _scheduleItemRequestRepository;
-
+    private readonly IDbRepository<ScheduleItem> _scheduleItemRepository;
+    private readonly IDbRepository<Ad> _adRepository;
     public TutorService(IDbRepository<ScheduleItem> scheduleItemRepository,
-                        IDbRepository<ScheduleItemRequest> scheduleItemRequestRepository)
+                        IDbRepository<ScheduleItemRequest> scheduleItemRequestRepository,
+                        IDbRepository<Ad> adRepository)
     {
         _scheduleItemRepository = scheduleItemRepository;
         _scheduleItemRequestRepository = scheduleItemRequestRepository;
+        _adRepository = adRepository;
+    }
+
+    public async Task<IsUserTheAdOwnerResponse> IsUserTheAdOwner(IsUserTheAdOwnerRequest request)
+    {
+        int adId = request.AdId;
+        int userId = (int)request.UserId;
+
+        Ad? ad = await _adRepository.GetById(adId);
+        bool isOwner = ad != null && ad.TutorId == userId;
+
+        return new IsUserTheAdOwnerResponse
+        {
+            IsOwner = isOwner,
+        };
     }
 
-    public Task<IsUserTheAdOwnerResponse> IsUserTheAdOwner(IsUserTheAdOwnerRequest request)
+    public async Task<CreateScheduleItemResponse> CreateItem(CreateScheduleItemRequest request)
     {
-        // TODO add logic (this is only for tests)
-        var response = new IsUserTheAdOwnerResponse
+        int adId = request.AdId;
+        int userId = request.UserId;
+        DateTime dateTime = request.DateTime;
+
+        Ad? ad = await _adRepository.GetById(adId);
+        bool isOwner = ad != null && ad.TutorId == userId;
+
+        if (!isOwner)
+        {
+            return new CreateScheduleItemResponse
+            {
+                Success = false
+            };
+        }
+
+        var lastScheduleItem = await _scheduleItemRepository.GetAll().OrderByDescending(x => x.Id).FirstOrDefaultAsync();
+        int lastItemId = lastScheduleItem?.Id ?? 0;
+
+        int createdItemId = lastItemId + 1;
+
+        var scheduleItem = new ScheduleItem(createdItemId, adId, dateTime);
+
+        await _scheduleItemRepository.Create(scheduleItem);
+
+        return new CreateScheduleItemResponse
         {
-            IsOwner = true
+            Success = true,
+            CreatedItemId = createdItemId
```

Comment:
After `Create()`, `scheduleItem` you provided it will have its `Id` set by `Db`, so you can use that. You might want to check if it's not 0 and send response with `Success = false` if it is, but most likely the create would throw an exception if anything went wrong, so we can skip it.